minari/dataset/_storages/arrow_storage.py

```diff
@@ -178,11 +180,20 @@ def _encode_space(space: gym.Space, values: Any, pad: int = 0):
         return pa.StructArray.from_arrays(arrays, names=names)
     elif isinstance(space, _FIXEDLIST_SPACES):
         values = np.asarray(values)
-        assert values.shape[1:] == space.shape
-        values = values.reshape(values.shape[0], -1)
-        values = np.pad(values, ((0, pad), (0, 0)))
-        dtype = pa.list_(pa.from_numpy_dtype(space.dtype), list_size=values.shape[1])
-        return pa.FixedSizeListArray.from_arrays(values.reshape(-1), type=dtype)
+        if values.shape == (4, 84, 84) and values.dtype == np.uint8: # check for image observation (4 stacked greyscale images)
```

Comment:
we should not constrain ourselves with 84 x 84 images, but we should accept any size.

Same reasoning for the 4 dimension.

I think the discriminant here is to have Box, with at least 2 dim, and with type uint8. Then you can put a logging.warn saying you are considering it as image, and if this is not intended to disable it via a flag "image_observation" which is defaulted to None. Then you can compute the value of the flag during init (and warn just once). I will clarify later in our meeting.